scripts/run_e2e_tests.py

```diff
@@ -0,0 +1,472 @@
+    if os.path.isfile(GECKO_PROVIDER_BAK_FILE_PATH):
+        os.remove(GECKO_PROVIDER_FILE_PATH)
+        os.rename(GECKO_PROVIDER_BAK_FILE_PATH, GECKO_PROVIDER_FILE_PATH)
+
+
+def start_webdriver_manager():
+    """Update and start webdriver manager."""
+    if common.is_windows_os():
+        tweak_webdriver_manager()
+
+    run_webdriver_manager(
+        ['update', '--versions.chrome', CHROME_DRIVER_VERSION])
+    run_webdriver_manager(
+        ['start', '--versions.chrome', CHROME_DRIVER_VERSION,
+         '--detach', '--quiet'])
+
+    if common.is_windows_os():
+        undo_webdriver_tweak()
+
+
+def get_parameter_for_config_file(run_on_browserstack):
+    """Return the parameter for the target configuration file based on whether
+    the test is running on browserstack.
+
+    Args:
+        run_on_browserstack: bool. Whether run the test on browserstack.
+
+    Returns:
+        str: The config file path.
+    """
+    if not run_on_browserstack:
+        return PROTRACTOR_CONFIG_FILE_PATH
+    else:
+        python_utils.PRINT('Running the tests on browsertack...')
+        return BROWSER_STACK_CONFIG_FILE_PATH
+
+
+def get_parameter_for_sharding(sharding, sharding_instances):
+    """Return the parameter for sharding. Based on the sharding instances
+
+    Args:
+        sharding: bool. Whether run the test with sharding.
+        sharding_instances: int. How many sharding instances to be running.
+
+    Returns:
+        list(str): A list of parameters to represent the sharding configuration.
+    """
+    if not sharding or sharding_instances == 1:
+        return []
+    else:
+        return ['--capabilities.shardTestFiles=%s' % sharding,
+                '--capabilities.maxInstances=%s' % sharding_instances]
+
+
+def get_parameter_for_dev_mode(dev_mode):
+    """Return parameter for whether the test should be running on dev_mode
+
+    Args:
+        dev_mode: bool. Whether the test is running on dev_mode.
+
+    Returns:
+        str: A string for the testing mode command line parameter.
+    """
+    return '--params.devMode=%s' % dev_mode
+
+
+def get_parameter_for_suite(suite):
+    """Return parameter for which suite to be running on the tests.
+
+    Args:
+        suite: str. The suite name to be running on the tests. If the value is
+            `full`, all tests will run.
+
+    Returns:
+        list(str): A list of command line parameter for suite.
+    """
+    return ['--suite', suite]
+
+
+def get_e2e_test_parameters(
+        run_on_browserstack, sharding, sharding_instances, suite, dev_mode):
+    """Return parameters for the end-2-end tests.
+
+    Args:
+        run_on_browserstack: bool. Represents whether the tests should run
+            on browserstack.
+        sharding: bool. Disables/Enables parallelization of protractor tests.
+        sharding_instances: str. Sets the number of parallel browsers to open
+            while sharding.
+        suite: str. Performs test for different suites.
```

Comment:
The name of the suite whose tests should be run.

(Also, maybe change this and similar args to suite_name.)

Reply:
Updated